src/devices/machine/i8291a.cpp

```diff
@@ -0,0 +1,1130 @@
+// license:BSD-3-Clause
+// copyright-holders: Sven Schnelle
+/*********************************************************************
+
+ i8291a.cpp
+
+ Intel 8291A GPIB controller
+
+ *********************************************************************/
+
+#include "emu.h"
+
+
+#include "i8291a.h"
+
+#define VERBOSE 0
+
+#include "logmacro.h"
+
+#define LOG_STATE_MASK  (LOG_GENERAL << 1)
+#define LOG_STATE(...)  LOGMASKED(LOG_STATE_MASK, __VA_ARGS__)
+#define LOG_REG_MASK    (LOG_STATE_MASK << 1)
+#define LOG_REG(...)    LOGMASKED(LOG_REG_MASK, __VA_ARGS__)
+#define LOG_INT_MASK    (LOG_REG_MASK << 1)
+#define LOG_INT(...)    LOGMASKED(LOG_INT_MASK, __VA_ARGS__)
+#define LOG_CMD_MASK	(LOG_INT_MASK << 1)
+#define LOG_CMD(...)	LOGMASKED(LOG_CMD_MASK, __VA_ARGS__)
+
+
+DEFINE_DEVICE_TYPE(I8291A, i8291a_device, "i8291a", "I8291A")
```

Comment:
Please use a more descriptive description (the second string argument).

Reply:
fixed

Reply:
fixed